### DISCUSSION TITLE
Does \GraphQL\Language\Parser::parse preserve comments from a source SDL?

### DISCUSSION THREAD

**Discussion by rhift** [webonyx/graphql-php#1460] [General]
If it does I can't get it to work for me hahaha =) any help would be very appreciated. thank you!

**Comment by spawnia** [comment]
No. It used to do it when comments were used as descriptions, but now it just discards them.

I am very suspicious of why this would be necessary or useful. Comments are just a vehicle of the source text, they do not clearly correlate to the parsed objects.

**Comment by rhift** [comment]
Ah, thanks! My use case is to allow the comments to be used with GraphiQL as documentation.

**Reply by spawnia** [reply]
There are descriptions for that.

**Comment by rhift** [comment]
How can I add a description to an SDL document?
```
# This is a description.
type Query {
}
```

**Reply by spawnia** [reply]
https://spec.graphql.org/October2021/#sec-Descriptions

**Reply by rhift** [reply]
Awesome, thanks @spawnia !
